salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                            for role in existing_roles:
+                                if role in {k.lower(): v for k, v in local_roles.items()}:
+                                    log.debug(
+                                        'AD role ["%s"] exists in configuration, no cleanup necessary', role)
+                                else:
+                                    log.debug(
+                                        'Ad role ["%s"] does not exists in configuration, deleting...', role)
+                                    client.secrets.activedirectory.delete_role(
+                                        name=role
+                                    )
+                                    log.debug(
+                                        'AD role has been ["%s"] deleted.', role)
+                else:
+                    log.debug(
+                        'Secret engine "%s" does not contain any extra configurations.', secret_engine.type
+                    )
+        except Exception:
+            raise
+
+        # Build return object
+        ret['changes']['old'] = ["type: {} - path: {}".format(ob.type, ob.path) for ob in remote_engines]
+        ret['changes']['new'] = new_secrets_engines or "No changes"
+
+        log.info('Finished proccessing and configuring secrets engines.')
+
+    def cleanup_secrets_engines(self, client, remote_engines, local_engines, ret):
+        """Disabling any secrets engines that are not present in the local config
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_engines {list} -- secrets engines from remote vault server
+            local_engines {list} -- secrets engines from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up secrets engines...')
+        has_changes = False
+
+        try:
+            for secret_engine in remote_engines:
+                if not (secret_engine.type == "system" or
+                        secret_engine.type == "cubbyhole" or
+                        secret_engine.type == "identity" or
+                        secret_engine.type == "generic"):
+                    if secret_engine in local_engines:
+                        log.debug('Secrets engine "%s" at path "%s" exists in configuration, no cleanup necessary.',
+                                  secret_engine.type, secret_engine.path)
+                    else:
+                        log.debug('Secrets engine "%s" at path "%s" does not exist in configuration. Disabling...',
+                                  secret_engine.type, secret_engine.path)
+                        has_changes = True
+                        client.sys.disable_secrets_engine(
+                            path=secret_engine.path
+                        )
+                        log.debug('Secrets engine "%s" at path "%s" is disabled.',
+                                secret_engine.type, secret_engine.type)
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.